HopsanGUI/LogVariable.cpp

```diff
@@ -805,6 +805,130 @@ SharedVectorVariableT VectorVariable::toFrequencySpectrum(const SharedVectorVari
     }
 }
 
+SharedVectorVariableT VectorVariable::identifySteadyState(const SteadyStateIdentificationMethodEnumT method, double tol, double win, double stdev, double l1, double l2, double l3)
+{
+    SharedVectorVariableT pRetVar = mpParentLogDataHandler->createOrphanVariable(this->getFullVariableName()+"_ss", TimeDomainType);
+
+    SharedVectorVariableT pTime = this->getSharedTimeOrFrequencyVector();
+    QVector<double> time = pTime->getDataVectorCopy();
+    QVector<double> data = this->getDataVectorCopy();
+
+    if(method == RectangularWindowTest) {
+        int nw = 0;
+        for(int i=0; i<time.size(); ++i) {
+            if(time[i] < win) {
+                pRetVar->append(time[i], 0);
+                nw = i+1;  //Update number of samples for window
+                continue;
+            }
+
+            //Compute first variance
+            double xmax = data[i-nw];
+            double xmin = data[i-nw];
+            for(int j=i-nw+1; j<i; ++j) {
```

Comment:
I am not exactly sure what is going on here, maybe it must be this way, but this loop in a loop risk becoming very slow for very large time vectors. Especially if i  nw is o low number.